src/lib.rs

```diff
@@ -1474,6 +1474,144 @@ fn collect_runs(
     Ok(out_list)
 }
 
+/// Collect runs that match a filter function
+///
+/// A bicolor run is a path of nodes connected by edges of up two colors,
```

Comment:
I wonder if *path* is the correct term here since a collected block may have nodes with `out_degree > 1` and even be "disconnected" since if `filter_node` returns `None`, it doesn't "stop" the collection of a block.

Reply:
I went with list of group of nodes, and defined group as nodes connected by edges of exactly two colors.